code/modules/events/high_priority_bounty.dm

```diff
@@ -5,16 +5,29 @@
 	weight = 20
 	earliest_start = 10
 
+/datum/round_event
+	var/datum/bank_account/account
+
+/datum/round_event/high_priority_bounty/setup()
+	var/list/datum/bank_account/candidates
+
+	for(var/account_code as anything in SSeconomy.bank_accounts)
+		var/datum/bank_account/B = SSeconomy.bank_accounts[account_code]
+		if(!B) continue
+		if(!B.bounties) continue
+		if(B.bounties.len > 0)
+			candidates += B
+	
+	account = pick(candidates)
+
 /datum/round_event/high_priority_bounty/announce(fake)
-	priority_announce("Central Command has issued a high-priority cargo bounty. Details have been sent to all bounty consoles.", "Nanotrasen Bounty Program")
+	priority_announce("Central Command has issued a high-priority cargo bounty to [account.account_holder]. Details have been sent to all bounty consoles.", "Nanotrasen Bounty Program")
 
 /datum/round_event/high_priority_bounty/start()
-	var/datum/bounty/B
-	for(var/attempts = 0; attempts < 50; ++attempts)
-		B = random_bounty()
-		if(!B)
-			continue
-		B.mark_high_priority(3)
-		if(try_add_bounty(B))
-			break
+	var/datum/bounty/B = random_bounty(CIV_JOB_RANDOM, account)
```

Comment:
Shouldn't it generate based on your job, rather than random?

Reply:
Its a more valuable bounty, so it should be harder to get